### DISCUSSION TITLE
Benchmark

### DISCUSSION THREAD

**Discussion by epezent** [epezent/implot#182] [General]
The demo now includes a new benchmark tool that you can use to gauge how many lines/points you can plot before your frame rate tanks. It can be accessed from the Tools menu bar. The tool ramps from 0 to 500 lines, each 1,000 pts. You can enable/disable anti-aliased lines to see what effect that has on performance. 

Here are a few runs I did. The first three are w/o AA, the last three with AA. My laptop specs:

Intel i7 9750H
16 GB RAM
Nvidia RTX 2080 Max-Q

If you have time, please consider posting your PC specs and benchmark results! 

![image](https://user-images.githubusercontent.com/29577475/91510226-0d144680-e8a2-11ea-8595-f67200af2809.png)

**Comment by ozlb** [comment]
2.26 GHz Intel Core 2 Duo (mid 2009)
8 GB RAM
NVIDIA GeForce 9400M 256 MB

OpenGL 3.2, ImDrawIdx == 32 bit, RenderereHasVtxOffset == true

![Screenshot 2020-08-28 at 16 54 51](https://user-images.githubusercontent.com/3250876/91581911-67b7ac00-e94f-11ea-89ad-aed143837d92.png)

200 lines (1000 pts each), the first three are w/o AA, the last with AA.

**Comment by pthom** [comment]
Webasm Emscripten version,  running on a MacBook Pro (16-inch, 2019), 2,4 GHz 8-Core Intel Core i9, 32 GB 2667 MHz DDR4, Graphics card Intel UHD Graphics 630 1536 MB:

<img width="276" alt="image" src="https://user-images.githubusercontent.com/7694091/92418983-eb9d3f80-f16a-11ea-9509-bcf6b805db46.png">


| Safari | Firefox |
|------ |--------|
|   <img width="501" alt="image" src="https://user-images.githubusercontent.com/7694091/92418936-9bbe7880-f16a-11ea-9328-9223196f7b4e.png">        |    <img width="509" alt="image" src="https://user-images.githubusercontent.com/7694091/92418964-babd0a80-f16a-11ea-905f-7a0ea436bdd6.png"> |

This was benchmarked using the emscripten interactive demo (where I had forcefully set the max fps to 200, instead of the preferrend fps chosen by the browser). 

(Note: the online demo will not allow to get fps higher than 50, since it uses the browser preferred fps)

**Comment by epezent** [comment]
@pthom, this post should be relevant: https://huww98.github.io/TimeChart/docs/performance
